File: Contest on Code Forces/May 25, 2024/main1.py
test = int(input())
for k in range(test):
    num = int(input())
    vals1 = list(map(int,input().split()))
    vals = vals1.copy()
    vals1.sort()

    ans  = list()
    n = len(vals)
    flag = False
    for i in range(len(vals)):
        ans = vals[i:] + vals[:i]
        ans +=ans
        index = 0
        flag = False
        for j in range(len(ans)):
            if(vals1[index]==ans[j]):
                index+=1
                if(index == len(vals)):
                    flag = True
                    break
            else:
                index = 0
                
            
           
        if(flag):
            print("YES")
            break

    if flag==False:
        print("NO")


    
    
    

File: Contest on Code Forces/August 13 2024/main1.cpp
# include <bits/stdc++.h>
using namespace std;
int main(){
    int test;
    cin>>test;
    while(test--){
        int num;
        cin>>num;
        if(num>=102 && num<=109)
        {
            cout<<"YES\n";
        }
        else if(num>=1010 && num<=1099){
            cout<<"YES\n";
        }
        else{
            cout<<"NO\n";
        }

    }
}